src/basic/efivars.h

```diff
@@ -12,20 +12,82 @@
 #define EFI_VARIABLE_NON_VOLATILE       0x0000000000000001
 #define EFI_VARIABLE_BOOTSERVICE_ACCESS 0x0000000000000002
 #define EFI_VARIABLE_RUNTIME_ACCESS     0x0000000000000004
 
+#define EFI_VENDOR_VARIABLE(vendor, name) name "-" vendor
+
+/* Define the variable. Keep original capitalization to make to avoid confusion. */
+#define EFI_VARIABLE_BootOrder \
+        EFI_VENDOR_VARIABLE(EFI_VENDOR_GLOBAL_STR, "BootOrder")
+#define EFI_VARIABLE_OsIndicationsSupported \
+        EFI_VENDOR_VARIABLE(EFI_VENDOR_GLOBAL_STR, "OsIndicationsSupported")
+#define EFI_VARIABLE_OsIndications \
+        EFI_VENDOR_VARIABLE(EFI_VENDOR_GLOBAL_STR, "OsIndications")
+#define EFI_VARIABLE_SecureBoot \
+        EFI_VENDOR_VARIABLE(EFI_VENDOR_GLOBAL_STR, "SecureBoot")
+#define EFI_VARIABLE_SetupMode \
+        EFI_VENDOR_VARIABLE(EFI_VENDOR_GLOBAL_STR, "SetupMode")
+
+#define EFI_VARIABLE_LoaderBootCountPath \
+        EFI_VENDOR_VARIABLE(EFI_VENDOR_LOADER_STR, "LoaderBootCountPath")
+#define EFI_VARIABLE_LoaderConfigTimeout \
+        EFI_VENDOR_VARIABLE(EFI_VENDOR_LOADER_STR, "LoaderConfigTimeout")
+#define EFI_VARIABLE_LoaderConfigTimeoutOneShot \
+        EFI_VENDOR_VARIABLE(EFI_VENDOR_LOADER_STR, "LoaderConfigTimeoutOneShot")
+#define EFI_VARIABLE_LoaderDevicePartUUID \
+        EFI_VENDOR_VARIABLE(EFI_VENDOR_LOADER_STR, "LoaderDevicePartUUID")
+#define EFI_VARIABLE_LoaderEntries \
+        EFI_VENDOR_VARIABLE(EFI_VENDOR_LOADER_STR, "LoaderEntries")
+#define EFI_VARIABLE_LoaderEntryOneShot \
+        EFI_VENDOR_VARIABLE(EFI_VENDOR_LOADER_STR, "LoaderEntryOneShot")
+#define EFI_VARIABLE_LoaderEntryDefault \
+        EFI_VENDOR_VARIABLE(EFI_VENDOR_LOADER_STR, "LoaderEntryDefault")
+#define EFI_VARIABLE_LoaderEntrySelected \
+        EFI_VENDOR_VARIABLE(EFI_VENDOR_LOADER_STR, "LoaderEntrySelected")
+#define EFI_VARIABLE_LoaderFeatures \
+        EFI_VENDOR_VARIABLE(EFI_VENDOR_LOADER_STR, "LoaderFeatures")
+#define EFI_VARIABLE_LoaderFirmwareInfo \
+        EFI_VENDOR_VARIABLE(EFI_VENDOR_LOADER_STR, "LoaderFirmwareInfo")
+#define EFI_VARIABLE_LoaderFirmwareType \
+        EFI_VENDOR_VARIABLE(EFI_VENDOR_LOADER_STR, "LoaderFirmwareType")
+#define EFI_VARIABLE_LoaderImageIdentifier \
+        EFI_VENDOR_VARIABLE(EFI_VENDOR_LOADER_STR, "LoaderImageIdentifier")
+#define EFI_VARIABLE_LoaderInfo \
+        EFI_VENDOR_VARIABLE(EFI_VENDOR_LOADER_STR, "LoaderInfo")
+#define EFI_VARIABLE_LoaderSystemToken \
+        EFI_VENDOR_VARIABLE(EFI_VENDOR_LOADER_STR, "LoaderSystemToken")
+#define EFI_VARIABLE_LoaderTimeInitUSec \
+        EFI_VENDOR_VARIABLE(EFI_VENDOR_LOADER_STR, "LoaderLoaderTimeInitUSec")
+#define EFI_VARIABLE_LoaderTimeExecUSec \
+        EFI_VENDOR_VARIABLE(EFI_VENDOR_LOADER_STR, "LoaderLoaderTimeExecUSec")
+#define EFI_VARIABLE_LoaderRandomSeed \
+        EFI_VENDOR_VARIABLE(EFI_VENDOR_LOADER_STR, "LoaderRandomSeed")
+#define EFI_VARIABLE_StubInfo \
+        EFI_VENDOR_VARIABLE(EFI_VENDOR_LOADER_STR, "StubInfo")
```

Comment:
hmm, now we have to maintain a header file with all variables we access? I am getting less and less convinced this is a good idea...

Also, you split efivars.h into efi-loader.h a while back, but most of these variables are only used by the latter, not the former. Something is askew with the split there.

(in fact, the split as it is implemented is already a bit skewed in itself, since efi-loader.h contains a bunch of functions that just expose firmware details and others which expose our loader variables. The former should probably be in a different file, since they are defined in the official EFI specs, while the latter are defined by use, and are defined in our boot loader spec)

Reply:
I think it makes sense to keep the variable declarations together. We *could* move some of them to a different header, but I think it would just make adding new variables a bit more awkward.

So yeah, the split into `efi-loader.[ch]` and `efivars.[ch]` is unfortunate, but it's caused by the desire to keep some code out of basic/, with the simultaneous need to read SystemdOptions in src/basic/ for other things which need systemd config in src/basic/. I looked at the files, and I don't see a particularly better way to arrange all this.